exec/selected/selected.go

```diff
@@ -55,6 +55,60 @@ type SchemaField struct {
 	FixedResult reflect.Value
 }
 
+func (sf *SchemaField) ToSelectedField() *types.SelectedField {
+	if sf == nil {
+		return nil
+	}
+	return &types.SelectedField{
+		Name:       sf.Name,
+		Alias:      sf.Alias,
+		Fields:     selsToSelectedFields(sf.Sels),
+		Args:       sf.Args,
+		Directives: sf.Directives,
+	}
+}
+
+func selsToSelectedFields(sels []Selection) (fs []*types.SelectedField) {
+	for _, f := range sels {
+		// Selection is SchemaField or TypeAssertion or TypenameField
+		switch v := f.(type) {
+		case *SchemaField:
+			fs = append(fs, v.ToSelectedField())
+		case *TypeAssertion:
+			/*
```

Comment:
just say that TypeAssertion is a selection dependent on the type of a union field